sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -304,102 +304,105 @@ int gnrc_ipv6_nib_get_next_hop_l2addr(const ipv6_addr_t *dst,
 
     DEBUG("nib: get next hop link-layer address of %s%%%u\n",
           ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)),
-          (netif != NULL) ? (unsigned)netif->pid : 0U);
+          netif ? netif->pid : 0);
```

Comment:
```suggestion
          netif ? (unsigned)netif->pid : 0U);
```

This is actually needed to ensure that the `%u` does match. You can go for `"... address of %s%%%" PRIkernel_pid "\n"` instead, but that will not print the number on `newlib_nano` (until `mpaland-printf` is used to replace newlib's printf implementation).